packages/@o3r/styling/scss/theming/_material.scss

```diff
@@ -0,0 +1,207 @@
+    border-style: solid,
+    medium-border-width: 2px,
+    thick-border-width: 4px,
+    light-separator-color: #AAA,
+    light-separator-style: solid,
+    light-separator-width: 1px,
+    separator-color: #AAA,
+    separator-style: solid,
+    separator-width: 2px,
+    panel-border: none,
+    // Radius
+    container-radius: 10px,
+    button-radius: 50%,
+    outline-radius: 0,
+    list-border-radius: 0,
+    // Shadow
+    enable-shadow-on-elements: false,
+    enable-shadow-on-containers: true,
+    shadow-color: rgba(0, 0, 0, 0.22),
+    // Text - theme
+    text: if($is-dark-theme, #FFF, #000)
+  );
+
+@if($is-dark-theme) {
+  $default: map.merge($default,
+      (primary-background: #050505,
+        progress-filter: linear-gradient(to left, rgba(255, 255, 255, 0.45), rgba(255, 255, 255, 0)),
+        dark-graphical-line: #61688F,
+      ));
+}
+
+@return map.merge($default, $implementation-override);
+}
+
+;
+
+/// Revert the palette
+/// contrast values become default values
+/// default values become contrasts values
+@function revert-palette($palette) {
+  @return map.merge(map.get($palette, 'contrast'), (contrast: map.remove($palette, 'contrast')));
+}
+
+;
+
+/// Generate the theme map containing a set of palettes and variables specific for this application
+/// @access public
+/// @param {map} $primary [$otter-primary] palette generated via define-palette with the colors in the primary theme (forms)
+/// @param {map} $highlight [$otter-highlight] palette generated via define-palette with the colors for the navigation and the continue / back buttons
+/// @param {map} $accent [$otter-accent] palette generated via define-palette with the colors used to accentuate one element in the page
+/// @param {map} $warn [$otter-warn] palette generated via define-palette with the colors used in warning or error feedbacks
+/// @param {bool} $is-dark-theme
+/// @param {map} $application [()] a set of properties to override for one specific implementation
+/// @return {map} containing the palette maps and the otter-theme variables map
+@function generate-theme($primary, $highlight, $accent, $warn, $is-dark-theme, $application: ()) {
+
+  $mat-theme: (
+  );
+$theme-variables: generate-theme-variables($is-dark-theme, $application);
+
+$theme-color: (
+  color: (primary: $primary,
+    accent: $accent,
+    warn: $warn,
+  )
+);
+
+@if ($is-dark-theme) {
+  $mat-theme: mat.define-dark-theme($theme-color);
+}
+
+@else {
+  $mat-theme: mat.define-light-theme($theme-color);
+}
+
+@return map.merge($mat-theme,
+  (highlight: $highlight,
+    application: $theme-variables ))
+}
+
+;
+
+/// Convert a Material Design theme to an Otter theme
+/// @param $mat-theme Material Design theme
+/// @access private
+@function _meta-theme-to-otter($meta-theme, $root-name) {
+  @if (meta.type-of($meta-theme) !=map) {
+    @return (variable: $root-name, // Deprecated, this is not used in the final theme calculation, will be removed in Otter v12
```

Comment:
since the method is recursive, all the node should end in this block, does this mean the whole _meta-theme-to-otter is deprecated?

Reply:
Actually the comment moved up (because of stylelint reorganization.....).
Only the field `variable` is deprecated